packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -45,14 +45,15 @@ export function containsDynamicChildren(children: IRNode[]): boolean {
     return children.some((child) => isElement(child) && isDynamic(child));
 }
 
-export function shouldFlatten(children: IRNode[]): boolean {
+export function shouldFlatten(children: IRNode[], state: State): boolean {
     return children.some(
         (child) =>
             isElement(child) &&
             (isDynamic(child) ||
                 !!child.forEach ||
                 !!child.forOf ||
-                (isTemplate(child) && shouldFlatten(child.children)))
+                (state.renderMode === 'light' && child.tag === 'slot') ||
```

Comment:
As we are changing the flattening logic in the codegen for light DOM slotting, it would be great to add a test fixture capturing this in the template compiler.